shell/plugins/steve/schema.utils.js

```diff
@@ -1,5 +1,19 @@
 import { normalizeType } from '@shell/plugins/dashboard-store/normalize';
 
+/**
+ * Returns _id special field without mutating the function input
+ */
+export function _id(schema) {
```

Comment:
it's a bit hard to name that way because these functions aren't really adding the fields, they're returning the values, moreover the _addPreCacheFields method on the resourceCache uses the function name to map the values to the correct field on the resource.

I could change it to something like `_idMaker` and then adjust the logic in `_addPreCacheFields` so that I'm feeding it an array of objects with a `name` key and the function to return the value and then adjust how I'm feeding the names in in the schemaCache but I'm not certain that's the right move.

Reply:
I've changed `preCacheFields` from an array to map. 
- This allows other cache's to use specific functions for generic property names
  - in case of other objects also having `_id`, or have their own generic property that clashes iwth others
- The specific functions can then be named appropriately to their specific function